File: backend/src/main/java/com/NBE3_4_2_Team4/domain/asset/main/service/AssetService.java
package com.NBE3_4_2_Team4.domain.asset.main.service;

import com.NBE3_4_2_Team4.domain.asset.main.entity.AssetCategory;
import com.NBE3_4_2_Team4.domain.member.member.entity.Member;
import org.apache.commons.lang3.tuple.Pair;
import org.springframework.stereotype.Service;

@Service
public interface AssetService {
    public void transfer(String fromUsername, String toUsername, long amount, AssetCategory assetCategory);
    public Long deduct(String from, long amount, AssetCategory assetCategory);
    public Long accumulate(String to, long amount, AssetCategory assetCategory);
    public Long adminDeduct(String from, long amount, long admAstCategoryId);
    public Long adminAccumulate(String to, long amount, long admAstCategoryId);
}

File: backend/src/main/java/com/NBE3_4_2_Team4/domain/asset/main/service/AssetService.kt
package com.NBE3_4_2_Team4.domain.asset.main.service

import com.NBE3_4_2_Team4.domain.asset.main.entity.AssetCategory
import org.springframework.stereotype.Service

@Service
interface AssetService {
    fun transfer(fromUsername: String, toUsername: String, amount: Long, assetCategory: AssetCategory)
    fun deduct(from: String, amount: Long, assetCategory: AssetCategory): Long
    fun accumulate(to: String, amount: Long, assetCategory: AssetCategory): Long
    fun adminDeduct(from: String, amount: Long, admAstCategoryId: Long): Long
    fun adminAccumulate(to: String, amount: Long, admAstCategoryId: Long): Long
}
